dlt/common/runtime/telemetry.py

```diff
@@ -25,6 +26,10 @@ def start_telemetry(config: RuntimeConfiguration) -> None:
     if _TELEMETRY_STARTED:
         return
 
+    # telemetry uses threading
+    if not platform_supports_threading():
```

Comment:
my take: change `RuntimeConfiguration` `on_resolved` and disable telemetry there.